src/components/paster.ts

```diff
@@ -0,0 +1,201 @@
+import * as vscode from 'vscode'
+import * as path from 'path'
+import * as fs from 'fs'
+import * as csv from 'csv-parser'
+
+import { Extension } from '../main'
+
+export class Paster {
+    extension: Extension
+
+    constructor(extension: Extension) {
+        this.extension = extension
+    }
+
+    public async paste() {
+        this.extension.logger.addLogMessage('Performing formatted paste')
+
+        // get current edit file path
+        const editor = vscode.window.activeTextEditor
+        if (!editor) {
+            return
+        }
+
+        const fileUri = editor.document.uri
+        if (!fileUri) {
+            return
+        }
+        if (fileUri.scheme === 'untitled') {
+            vscode.window.showInformationMessage('Before paste image, you need to save current edit file first.')
```

Comment:
Suggested message instead : "You need to the save the current editor before pasting image."

Reply:
We don't actually need this restriction any more because most formatted pastes won't need the file to be saved to a location (images being an exception) so I've tweaked that in f29b86e391b2398b60d95276ecf85257fed2e3e1